hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestCreateTable.scala

```diff
@@ -272,4 +277,154 @@ class TestCreateTable extends TestHoodieSqlBase {
       )
     }
   }
+
+  test("Test Create Table From Exist Hoodie Table") {
+    withTempDir { tmp =>
+      // Write a table by spark dataframe.
+      Seq("2021-08-02", "2021/08/02").foreach { partitionValue =>
+        val tableName = generateTableName
+        val tablePath = s"${tmp.getCanonicalPath}/$tableName"
+        import spark.implicits._
+        val df = Seq((1, "a1", 10, 1000, partitionValue)).toDF("id", "name", "value", "ts", "dt")
```

Comment:
thanks for expanding the test scope. 
In general, there are two types of partitioning. Simple and multi-level. Can you please add a test for multi-level as well. For now, we are manually testing these to ensure fix looks good. We should be able to cover all these via unit tests itself.

Reply:
done!